pkg/ingester/ingester.go

```diff
@@ -2443,7 +2443,8 @@ func (i *Ingester) shipBlocks(ctx context.Context, allowed *util.AllowedTenants)
 				level.Error(i.logger).Log("msg", "failed to update cached shipped blocks after shipper synchronisation", "user", userID, "err", err)
 			}
 		}
-
+		// reset shards for tenant table
+		i.limiter.ClearShardsForTenant()
```

Comment:
This introduces a data race, so I'll have to re-think locking around this map in the limiter.

Reply:
This was actually in the wrong place and should not have been in the per user loop